src/memray/_memray/hooks.h

```diff
@@ -26,22 +31,39 @@
 #    define MEMRAY_PLATFORM_HOOKED_FUNCTIONS
 #endif
 
-#define MEMRAY_HOOKED_FUNCTIONS                                                                         \
-    FOR_EACH_HOOKED_FUNCTION(malloc)                                                                    \
-    FOR_EACH_HOOKED_FUNCTION(free)                                                                      \
-    FOR_EACH_HOOKED_FUNCTION(calloc)                                                                    \
-    FOR_EACH_HOOKED_FUNCTION(realloc)                                                                   \
-    FOR_EACH_HOOKED_FUNCTION(posix_memalign)                                                            \
-    FOR_EACH_HOOKED_FUNCTION(aligned_alloc)                                                             \
-    FOR_EACH_HOOKED_FUNCTION(memalign)                                                                  \
-    FOR_EACH_HOOKED_FUNCTION(valloc)                                                                    \
-    FOR_EACH_HOOKED_FUNCTION(dlopen)                                                                    \
-    FOR_EACH_HOOKED_FUNCTION(dlclose)                                                                   \
-    FOR_EACH_HOOKED_FUNCTION(mmap)                                                                      \
-    FOR_EACH_HOOKED_FUNCTION(munmap)                                                                    \
-    FOR_EACH_HOOKED_FUNCTION(prctl)                                                                     \
-    FOR_EACH_HOOKED_FUNCTION(PyGILState_Ensure)                                                         \
-    MEMRAY_PLATFORM_HOOKED_FUNCTIONS
+#if defined(__linux__)
+#    define MEMRAY_HOOKED_FUNCTIONS                                                                     \
```

Comment:
The platform-specific differences are supposed to be captured by `MEMRAY_PLATFORM_HOOKED_FUNCTIONS` above, rather than making `MEMRAY_HOOKED_FUNCTIONS` platform specific and listing all the common ones twice. We used to only have 2 cases for `MEMRAY_PLATFORM_HOOKED_FUNCTIONS` (glibc vs musl), now we're adding a 3rd (Apple).

Looks like it should be:
```cpp
#if defined(__APPLE__)
#    define MEMRAY_PLATFORM_HOOKED_FUNCTIONS
#elif defined(__GLIBC__)
#    define MEMRAY_PLATFORM_HOOKED_FUNCTIONS                                                            \
        FOR_EACH_HOOKED_FUNCTION(memalign)                                                              \
        FOR_EACH_HOOKED_FUNCTION(prctl)                                                                 \
        FOR_EACH_HOOKED_FUNCTION(pvalloc)                                                               \
        FOR_EACH_HOOKED_FUNCTION(mmap64)
#else
#    define MEMRAY_PLATFORM_HOOKED_FUNCTIONS                                                            \
        FOR_EACH_HOOKED_FUNCTION(memalign)                                                              \
        FOR_EACH_HOOKED_FUNCTION(prctl)
#endif
```

If we want to split "platform" into 2 separate dimensions (OS vs libc) we can do that as well.

Reply:
Done